config/config.go

```diff
@@ -85,18 +90,23 @@ func Load(configFilePath string) (Config, error) {
 	}
 
 	file, err := os.OpenFile(configFilePath, os.O_RDONLY, 0644)
+
 	if err != nil {
 		if os.IsNotExist(err) {
 			return config, fmt.Errorf("could not find config file at path %s", configFilePath)
 		}
 		return config, err
 	}
-	defer file.Close()
 
 	if err := json.NewDecoder(file).Decode(&config); err != nil && !errors.Is(err, io.EOF) {
```

Comment:
Just a thought, if you are anyways reading the whole file, can you read it once as []byte and then use it for unmarshalling wtith Unmarshall (instead of Decode) and also compute the hash without the need for rewind?

Reply:
implementation updated, thanks!